samples/snippets/src/main/java/com/example/spanner/DirectedReadSample.java

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+// [START spanner_directed_read]
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Options;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.spanner.v1.DirectedReadOptions;
+import com.google.spanner.v1.DirectedReadOptions.ExcludeReplicas;
+import com.google.spanner.v1.DirectedReadOptions.IncludeReplicas;
+import com.google.spanner.v1.DirectedReadOptions.ReplicaSelection;
+
+public class DirectedReadSample {
+  static void directedRead() {
+    // TODO(developer): Replace these variables before running the sample.
+    final String projectId = "my-project";
+    final String instanceId = "my-instance";
+    final String databaseId = "my-database";
+    directedRead(projectId, instanceId, databaseId);
+  }
+
+  static void directedRead(String projectId, String instanceId, String databaseId) {
+    // Only one of excludeReplicas or includeReplicas can be set
+    // Each accepts a list of replicaSelections which contains location and type
+    //   * `location` - The location must be one of the regions within the
+    //      multi-region configuration of your database.
+    //   * `type` - The type of the replica
+    // Some examples of using replicaSelectors are:
+    //   * `location:us-east1` --> The "us-east1" replica(s) of any available type
+    //                             will be used to process the request.
+    //   * `type:READ_ONLY`    --> The "READ_ONLY" type replica(s) in nearest
+    // .                            available location will be used to process the
+    //                             request.
+    //   * `location:us-east1 type:READ_ONLY` --> The "READ_ONLY" type replica(s)
+    //                          in location "us-east1" will be used to process
+    //                          the request.
+    //  includeReplicas also contains an option called autoFailoverDisabled, which when set to true
+    //  Spanner will not route requests to a replica outside the
+    //  includeReplicas list when all the specified replicas are unavailable
+    //  or unhealthy. Default value is `false`.
+    final DirectedReadOptions directedReadOptionsForClient =
+        DirectedReadOptions.newBuilder()
+            .setExcludeReplicas(
+                ExcludeReplicas.newBuilder()
+                    .addReplicaSelections(
+                        ReplicaSelection.newBuilder().setLocation("us-east4").build())
+                    .build())
+            .build();
+
+    try (Spanner spanner =
```

Comment:
```suggestion
    // You can set default `DirectedReadOptions` for a Spanner client. These options will be applied to
    // all read-only transactions that are executed by this client, unless specific `DirectedReadOptions`
    // are set for a query.
    // Directed read can only be used for read-only transactions. The default options will be ignored for
    // any read/write transaction that the client executes.
    try (Spanner spanner =
```

Reply:
Thanks for suggesting this comment. This makes it more clear.